src/provider/provider_coarse.c

```diff
@@ -1643,9 +1642,7 @@ static umf_result_t coarse_memory_provider_allocation_merge(void *provider,
                                                             void *lowPtr,
                                                             void *highPtr,
                                                             size_t totalSize) {
-    if (provider == NULL || lowPtr == NULL || highPtr == NULL ||
-        ((uintptr_t)highPtr <= (uintptr_t)lowPtr) ||
-        ((uintptr_t)highPtr - (uintptr_t)lowPtr >= totalSize)) {
+    if (provider == NULL) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
```

Comment:
can we unify this for all funcionts perhaps...?
I can see `purge_force` functions check for null ptr and we should probably have this check in the upper layer (in `src/memory_provider.c`). The same e.g. in `ipc_handle_size` functions - all of them individually check if `size == null`... Does this make sense...?

Reply:
Done

Reply:
@lukaszstolarczuk ping

Reply:
@lukaszstolarczuk ping

Reply:
thx!